crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -1859,11 +1860,21 @@ impl<'a> Checker<'a> {
                         self.visit_parameters(parameters);
                     }
                     self.visit_expr(body);
+
+                    visited_lambdas.push((expr, snapshot));
```

Comment:
I don't expect you to accept that, but I couldn't figure out how to make it work otherwise :)

Reply:
Going to tweak this a bit, but the idea makes sense :)